document-generator-core/src/main/java/uk/gov/companieshouse/document/generator/core/avro/DocumentGenerationStateAvroSerializer.java

```diff
@@ -0,0 +1,62 @@
+package uk.gov.companieshouse.document.generator.core.avro;
+
+
+import uk.gov.companieshouse.document.generator.core.document.models.DocumentGenerationCompleted;
+import uk.gov.companieshouse.document.generator.core.document.models.DocumentGenerationFailed;
+import uk.gov.companieshouse.document.generator.core.document.models.DocumentGenerationStarted;
+
+import java.io.IOException;
+
+public class DocumentGenerationStateAvroSerializer {
+
+    /**
+     * Serialize the message for the completion of the document generation.
+     *
+     * @return
+     * @throws IOException
+     */
+    public byte[] serialize() throws IOException {
+        return serialize();
+    }
+
+    /**
+     * Serialize the message for the completion of the document generation.
+     *
+     * @param document
+     * @return
+     * @throws IOException
+     */
+    public byte[] serialize(DocumentGenerationCompleted document) throws IOException {
+        AvroSerializer<DocumentGenerationCompleted> serializer = new AvroSerializer<>();
+        return serializer.serialize(document);
+    }
+
+    /**
+     * Serialize the message for the start of the document generation.
+     *
+     * @param started
+     * @return
+     * @throws IOException
+     */
+    public byte[] serialize(DocumentGenerationStarted started) throws IOException {
+        AvroSerializer<DocumentGenerationStarted> serializer = new AvroSerializer<>();
+        return serializer.serialize(started);
+    }
+
+    /**
+     * Serialize the message for the document generation failure.
+     *
+     * The description values schema is a union with a null schema to
```

Comment:
This is revealing implementation detail of the schemas which is handled by Avro. If the schema changed this JavaDoc would become incorrect. JavaDocs should provide help for user's of the method, This JavaDoc is useful but it doesn't belong here.